Source/Icebreaker/Bot/IcebreakerBot.cs

```diff
@@ -314,6 +325,42 @@ private async Task HandleMessageActivityAsync(ITurnContext turnContext, Cancella
             }
         }
 
+         /// <summary>
+        /// Handle profile updates from users.
+        /// </summary>
+        /// <param name="activity">Message from submitted card</param>
+        /// <param name="turnContext">Context object containing information cached for a single turn of conversation with a user.</param>
+        /// <param name="cancellationToken">Propagates notification that operations should be canceled.</param>
+        /// <returns>Tracking Task</returns>
+        private async Task OnProfileSaveAsync(Activity activity, ITurnContext turnContext, CancellationToken cancellationToken)
+        {
+            var cardPayload = JToken.Parse(activity.Value.ToString());
+            var cardType = cardPayload["action"].Value<string>().ToLowerInvariant();
+
+            switch (cardType)
+            {
+                case "update":
+
+                    var profile = cardPayload["profile"].Value<string>();
+
+                    await this.dataProvider.SetUserProfileAsync(activity.From.Id, profile);
+
+                    // Confirmation message
+                    var reply = activity.CreateReply();
+                    reply.Attachments = new List<Attachment>
+                    {
+                        new HeroCard()
+                        {
+                            Text = "Your profile has been updated!"
+                        }.ToAttachment(),
+                    };
+
+                    await turnContext.SendActivityAsync(reply, cancellationToken).ConfigureAwait(false);
+
+                    break;
```

Comment:
Please add a default case with a warning logged on what type of action was attempted.